lib/internal/child_process/promises.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const util = require('util');
+const { getSystemErrorName } = require('internal/util');
+const { Buffer } = require('buffer');
+const {
+  ERR_CHILD_PROCESS_STDIO_MAXBUFFER,
+} = require('internal/errors').codes;
+const {
+  normalizeExecArgs,
+  sanitizeKillSignal,
+  spawn,
+  validateMaxBuffer,
+  validateTimeout
+} = require('internal/child_process/utils');
+
+async function exec() {
+  const opts = normalizeExecArgs.apply(null, arguments);
+  return execFile(opts.file, opts.options);
+}
+
+async function execFile(file /* , args, options */) {
```

Comment:
I think `async` is not needed here because this returns Promise object and `await` is not used. `exec()` is too.

```suggestion
function execFile(file /* , args, options */) {
```